zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -497,219 +518,201 @@ pub(super) const TABLE_SQLITE_SEQUENCE: &str = "CREATE TABLE sqlite_sequence(nam
 // Views
 //
 
-pub(super) const VIEW_RECEIVED_NOTES: &str = "
-CREATE VIEW v_received_notes AS
-SELECT
-    sapling_received_notes.id AS id_within_pool_table,
-    sapling_received_notes.tx,
-    2 AS pool,
-    sapling_received_notes.output_index AS output_index,
-    account_id,
-    sapling_received_notes.value,
-    is_change,
-    sapling_received_notes.memo,
-    sent_notes.id AS sent_note_id
-FROM sapling_received_notes
-LEFT JOIN sent_notes
-ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
-    (sapling_received_notes.tx, 2, sapling_received_notes.output_index)
+pub(super) const VIEW_RECEIVED_OUTPUTS: &str = "
+CREATE VIEW v_received_outputs AS
+    SELECT
+        sapling_received_notes.id AS id_within_pool_table,
+        sapling_received_notes.tx AS transaction_id,
+        2 AS pool,
+        sapling_received_notes.output_index,
+        account_id,
+        sapling_received_notes.value,
+        is_change,
+        sapling_received_notes.memo,
+        sent_notes.id AS sent_note_id
+    FROM sapling_received_notes
+    LEFT JOIN sent_notes
+    ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
+       (sapling_received_notes.tx, 2, sapling_received_notes.output_index)
 UNION
-SELECT
-    orchard_received_notes.id AS id_within_pool_table,
-    orchard_received_notes.tx,
-    3 AS pool,
-    orchard_received_notes.action_index AS output_index,
-    account_id,
-    orchard_received_notes.value,
-    is_change,
-    orchard_received_notes.memo,
-    sent_notes.id AS sent_note_id
-FROM orchard_received_notes
-LEFT JOIN sent_notes
-ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
-    (orchard_received_notes.tx, 3, orchard_received_notes.action_index)";
-
-pub(super) const VIEW_RECEIVED_NOTE_SPENDS: &str = "
-CREATE VIEW v_received_note_spends AS
+    SELECT
+        orchard_received_notes.id AS id_within_pool_table,
+        orchard_received_notes.tx AS transaction_id,
+        3 AS pool,
+        orchard_received_notes.action_index AS output_index,
+        account_id,
+        orchard_received_notes.value,
+        is_change,
+        orchard_received_notes.memo,
+        sent_notes.id AS sent_note_id
+    FROM orchard_received_notes
+    LEFT JOIN sent_notes
+    ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
+       (orchard_received_notes.tx, 3, orchard_received_notes.action_index)
+UNION
+    SELECT
+        u.id AS id_within_pool_table,
+        u.transaction_id,
+        0 AS pool,
+        u.output_index,
+        u.account_id,
+        u.value_zat AS value,
+        0 AS is_change,
```

Comment:
How is `is_change` used? `sapling_received_notes` and `orchard_received_notes` have columns `is_change INTEGER NOT NULL`; is it currently a glorified boolean (i.e. `0` specifically means "is not change"), or can it take other values? It would be useful to document this (either on the `sapling_received_notes` and `orchard_received_notes` tables, or on this view).

(I know that we set `0` for this in `v_transactions` below, but it would be good to document this.)

Reply:
There is no boolean type supported by sqlite, so we have to use an integer.

Reply:
That's what I figured; please document this.

Reply:
You can add a constraint: `CONSTRAINT foo_is_boolean CHECK (foo = FALSE OR foo = TRUE)`.

`FALSE` and `TRUE` are synonyms for 0 and 1, but using them makes the intent clear, and the constraint enforces it.

```suggestion
        FALSE AS is_change,
```

Reply:
Where we use this column, we actually use it as a count, so probably the better naming would be `change_count` or something: 
```        
SUM(notes.is_change) > 0     AS has_change,
```